### PR TITLE
feat(GaussDB): add gaussdb mysql account resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add gaussdb mysql account resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add gaussdb mysql account resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb/' TESTARGS='-run TestAccGaussDBAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccGaussDBAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBAccount_basic
=== PAUSE TestAccGaussDBAccount_basic
=== CONT  TestAccGaussDBAccount_basic
--- PASS: TestAccGaussDBAccount_basic (1080.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1195.000s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_mysql_account_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	getGaussDBAccountPath := getGaussDBAccountClient.Endpoint + getGaussDBAccountHttpUrl
+	getGaussDBAccountPath = strings.ReplaceAll(getGaussDBAccountPath, "{project_id}",
+		getGaussDBAccountClient.ProjectID)
+	getGaussDBAccountPath = strings.ReplaceAll(getGaussDBAccountPath, "{instance_id}", instanceID)
+
+	getGaussDBAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_mysql_account_test.go]: it is unnecessary to add okcode 200. Please check the other parts.

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+	if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok && errCode.Actual == 409 {
+		var apiError interface{}
+		if jsonErr := json.Unmarshal(errCode.Body, &apiError); jsonErr != nil {
+			return false, jsonErr
+		}
+		errorCode, errorCodeErr := jmespath.Search("error_code", apiError)
+		if errorCodeErr != nil {
+			return false, fmt.Errorf("error parse errorCode from response body: %s", errorCodeErr)
+		}
+		if errorCode == "DBS.200047" {
```
Comment (deer-hang) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go]: please add a description for this error code

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+	createGaussDBAccountRespBody, err := utils.FlattenResponse(createGaussDBAccountResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jobId, err := jmespath.Search("job_id", createGaussDBAccountRespBody)
+	if err != nil {
+		return diag.Errorf("error creating GaussDB MySQL account: job_id is not found in API response")
+	}
+	err = waitForJobComplete(ctx, createGaussDBAccountClient, d.Timeout(schema.TimeoutCreate), instanceID, jobId.(string))
```
Comment (deer-hang) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go]: this line is more than 120

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go [hunk 4]
```diff
@@ -0,0 +1,566 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jobId, err := jmespath.Search("job_id", deleteGaussDBAccountRespBody)
+	if err != nil {
+		return diag.Errorf("error deleting GaussDB MySQL account: job_id is not found in API response")
+	}
+
+	err = waitForJobComplete(ctx, deleteGaussDBAccountClient, d.Timeout(schema.TimeoutUpdate), instanceID, jobId.(string))
```
Comment (deer-hang) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go]:
```suggestion
	err = waitForJobComplete(ctx, deleteGaussDBAccountClient, d.Timeout(schema.TimeoutDelete), instanceID, jobId.(string))
```
this line is more than 120

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/gaussdb/' TESTARGS='-run  TestAccGaussDBAccount_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run  TestAccGaussDBAccount_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussDBAccount_basic
=== PAUSE TestAccGaussDBAccount_basic
=== CONT  TestAccGaussDBAccount_basic


--- PASS: TestAccGaussDBAccount_basic (832.02s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   832.086s

File: docs/resources/gaussdb_mysql_account.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  including letters, digits, and underscores (_).
+
+  Changing this parameter will create a new resource.
+
+* `password` - (Required, String) Specifies the password of the database user. It cannot be the same as the username.
+  The value cannot be empty and must consist of 8 to 32 characters and contain at least three types of the following:
+  uppercase letters, lowercase letters, digits, and special characters (~!@#$%^*-_=+?,()&).
+
+* `host` - (Optional, String, ForceNew) Specifies the host IP address. The default value is %, indicating that all IP
+  addresses are allowed to access your GaussDB(for MySQL) instance. If its value is 10.10.10. %, all 10.10.10.X IP
```
Comment (ShiChangkuo) [inline docs/resources/gaussdb_mysql_account.md]:
```suggestion
  addresses are allowed to access your GaussDB(for MySQL) instance. If its value is 10.10.10.%, all 10.10.10.X IP
```

File: docs/resources/gaussdb_mysql_account.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+$ terraform import huaweicloud_gaussdb_mysql_account.test <instance_id>/<name>/<host>
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include: `password`. It is generally recommended
+running `terraform plan` after importing a cluster. You can then decide if changes should be applied to the GaussDB
+MySQL account, or the resource definition should be updated to align with the GaussDB MySQL account. Also you can
+ignore changes as below.
+
+```
```
Comment (ShiChangkuo) [inline docs/resources/gaussdb_mysql_account.md]:
```suggestion
```hcl
```

**Pull request merged**